### DISCUSSION TITLE
searching behavior in the data viewer

### DISCUSSION THREAD

**Discussion by junichiyamasaki** [REditorSupport/vscode-R#1066] [Q&A] [answered]
Hi, I am a kind of data scientist and trying to switch from Rstudio to vscode-R, which I find more comfortable in most cases. I appreciate all the efforts made by the contributors.

One thing I noticed is the different behavior of data viewer. In RStudio, when we search for words in the data viewer using cmd+F, it only shows rows that have some matched columns. This is so convenient particularly when you search for particular words in a large data frame. 

On the other hand, when we search in vscode-R, it shows the search results with keeping unmatched rows. Therefore, to find the matched rows, we have to check all pages by clicking the page button even if the matched rows are just a few. Of course, we can do it by code like
`df %>% dplyr::filter(stringr::str_detect(var, "words")) %>% View()
`
, but sometimes the search window in the interactive data viewer is handier to work with. Is there any way to mimic R the Studio-like behavior in vscode-R? For example, is there any way to increase the number of rows per page? If we can set it to infinity, it automatically answers my question.

**Comment by renkun-ken** [comment, marked as answer]
When you press cmd+F, it triggers the webpage search for text in the tab.

Looks like you need data filter? Please check out the filter button on the right of each column:

<img width="460" alt="image" src="https://user-images.githubusercontent.com/4662568/162567335-c1b6bc30-7adb-4047-8e6f-0de26cdd80e6.png">

**Reply by junichiyamasaki** [reply]
Thank you very much for the suggestion.  The data filter works well, but sometimes we want to search in multiple columns, which is when I feel cmd+F is more convinient.

**Reply by renkun-ken** [reply]
Yes, it is definitely a valid use case we don't have good support yet.

As for customizing the number of rows per page, it is allowed via https://www.ag-grid.com/angular-data-grid/row-pagination/#example-customising-pagination, but we don't expose such an option to user at the moment. We could put it as a vscode setting. If we put the default dropdown box on top of the data grid, the web page layout will not look good.

**Reply by junichiyamasaki** [reply]
Thank you for elaboration. I will use data filter as you suggested for the time being, which will be sufficient in many cases.

**Reply by renkun-ken** [reply]
#1068 adds `r.session.data.pageSize` to the vscode settings. Using `0` will disable pagination and the default becomes 500.
